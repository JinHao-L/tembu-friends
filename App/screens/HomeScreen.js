import React, { Component } from 'react';
import { StyleSheet, SafeAreaView, ActivityIndicator, View } from 'react-native';
import WebView from 'react-native-webview';
import { Button } from 'react-native-elements';

import { Colors } from '../constants/index';
import Layout from '../constants/Layout';
import { MAIN_FONT, MainText } from '../components';

const WebNavButton = ({ title, disabled, onPress }) => {
    return (
        <Button
            containerStyle={{ flex: 1, borderRadius: 0 }}
            buttonStyle={{
                backgroundColor: Colors.appGreen,
                borderRadius: 0,
                borderWidth: StyleSheet.hairlineWidth,
                borderColor: Colors.appWhite,
            }}
            title={title}
            titleStyle={{ fontFamily: MAIN_FONT, fontSize: 15, color: Colors.appWhite }}
            onPress={onPress}
            disabledStyle={{ backgroundColor: Colors.appGray1 }}
            disabledTitleStyle={{ color: Colors.appGray2 }}
            disabled={disabled}
        />
    );
};

class HomeScreen extends Component {
    state = {
        canGoBack: false,
        canGoForward: false,
        currentUrl: 'https://tembusu.nus.edu.sg/',
    };
    webviewRef = React.createRef();
    homepage = 'https://tembusu.nus.edu.sg/';

    backButtonHandler = () => {
        if (this.state.canGoBack) {
            this.webviewRef.current.goBack();
        }
    };

    homeButtonHandler = () => {
        this.setState({ currentUrl: this.homepage });
        // this.webviewRef.current.injectJavaScript('window.location="' + this.homepage + '"');
    };

    frontButtonHandler = () => {
        if (this.state.canGoForward) {
            this.webviewRef.current.goForward();
        }
    };

    render() {
        const { canGoForward, canGoBack } = this.state;
        return (
            <SafeAreaView style={styles.container}>
                <WebView
                    source={{ uri: this.state.currentUrl }}
                    style={{ width: Layout.window.width }}
                    renderLoading={this.renderLoading}
                    startInLoadingState={true}
                    domStorageEnabled={true}
                    javaScriptEnabled={true}
                    ref={this.webviewRef}
                    onNavigationStateChange={(navState) => {
                        this.setState({
                            canGoBack: navState.canGoBack,
                            canGoForward: navState.canGoForward,
                            currentUrl: navState.url,
                        });
                    }}
                />
                <View style={styles.tabBarContainer}>
                    <WebNavButton
                        onPress={this.backButtonHandler}
                        disabled={!canGoBack}
                        title={'Back'}
                    />
                    <WebNavButton onPress={this.homeButtonHandler} title={'Home'} />
                    <WebNavButton
                        onPress={this.frontButtonHandler}
                        disabled={!canGoForward}
                        title={'Forward'}
                    />
                </View>
            </SafeAreaView>
        );
    }
}

function renderLoading() {
    return <ActivityIndicator size="large" />;
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: Colors.appWhite,
        alignItems: 'center',
    },
    contentContainer: {
        paddingTop: 30,
    },
    tabBarContainer: {
        flexDirection: 'row',
    },
    button: {
        color: Colors.appGreen,
        fontSize: 18,
    },
    disabledButton: {
        color: Colors.appGray2,
        fontSize: 18,
    },
});

export default HomeScreen;
