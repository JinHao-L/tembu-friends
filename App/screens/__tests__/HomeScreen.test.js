import renderer from 'react-test-renderer';
import HomeScreen from '../HomeScreen';
import React from 'react';

describe('<HomeScreen/>', () => {
    jest.mock('react-native-webview', () => 'WebView');

    it('has 1 child', () => {
        const tree = renderer.create(<HomeScreen />).toJSON();
        expect(tree.children.length).toBe(2);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<HomeScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
