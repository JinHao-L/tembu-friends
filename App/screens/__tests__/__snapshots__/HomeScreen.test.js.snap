// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onNavigationStateChange={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "https://tembusu.nus.edu.sg/",
        }
      }
      startInLoadingState={true}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "width": 750,
          },
        ]
      }
      useSharedProcessPool={true}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 0,
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F1F1F1",
              "borderColor": "#FFFFFF",
              "borderLeftWidth": 0.5,
              "borderRadius": 0,
              "borderRightWidth": 0.5,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#D9D9D9",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 15,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 0,
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#248458",
              "borderColor": "#FFFFFF",
              "borderLeftWidth": 0.5,
              "borderRadius": 0,
              "borderRightWidth": 0.5,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 15,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Home
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 0,
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F1F1F1",
              "borderColor": "#FFFFFF",
              "borderLeftWidth": 0.5,
              "borderRadius": 0,
              "borderRightWidth": 0.5,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#D9D9D9",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 15,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Forward
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
